### ISSUE TITLE
Cataloging python packages errors out if "top_level.txt" entry is missing

### ISSUE THREAD

**Issue by wagoodman** [anchore/syft#259]
**What happened**:
As [reported by Wagde Zabit](https://anchorecommunity.slack.com/archives/C4QV1RX7Y/p1605029393066600)
```
[0250] ERROR failed to catalog: 1 error occurred:
	* unable to catalog python package=/usr/local/lib/python3.8/site-packages/ptyprocess-0.6.0.dist-info/METADATA: missing python package top_level.txt (package="/usr/local/lib/python3.8/site-packages/ptyprocess-0.6.0.dist-info/METADATA"
```
**What you expected to happen**:
Cataloging should have continued to conclusion with results instead of stopping 

**Environment**:
This was discovered with grype v0.3.0

**Notes**
This is where the problem is https://github.com/anchore/syft/blob/de2e6a13b8138eb751ee87033985dd7e6c39c4fe/syft/cataloger/python/package_cataloger.go#L125-L127 , and should probably only be a log statement instead of returning an error (a best-effort strategy).

**Comment by wagde-orca** [comment]
is it fixed in grype or not yet?

**Comment by luhring** [comment]
@wagde-orca We're actually working on that right this second! We fixed one more thing in Syft, and then will update Grype to include the fix.

**Comment by luhring** [comment]
@wagde-orca Hi there, we just released [grype v0.4.0](https://github.com/anchore/grype/releases/tag/v0.4.0) which should resolve this issue. Let us know if you have any trouble!
